File: benchmark/fp.f90
	program fp
	    implicit none
	    real*8 ans
            real*8 tmp
  	    real*8 eps
   	    real*8 x
    	    integer*8 count

    	    ans = 1
	    tmp = 1
	    eps = 1
	    x = -19.5
	    count = 0
   
    	    do while (eps > 1e-15)
     		count = count + 1
        	tmp = tmp * (x/count)
        	eps = abs(tmp)
        	ans = ans + tmp
                ! write(*,*) "count = ",count,"eps = ",eps
	    end do
    
    	    write(*,*) "exp(x) = ", exp(x)
  	    write(*,*) "ans = ", ans
	end program

File: benchmark/fp.c
#include <stdio.h>
#include <math.h>
#include <fenv.h>

int main(void)
{
    double ans = 1., tmp = 1;
    double eps = 1;
    double x = -19.5;
    long count = 0;

    printf("rounding mode = %d\n", fegetround());
    fesetround(FE_UPWARD);
    printf("rounding mode = %d\n", fegetround());
   
    while (eps > 1e-15) {
        count++;
        tmp *= x/(count);
        eps = fabs(tmp);
        ans += tmp;
//        printf("count = %d, eps = %.8le\n", count, eps);
    }
    
    printf("math.h exp(%.6lf) = %.9le\n", x, exp(x));
    printf("ans = %.16le\n", ans);

    return 0;
}

